packages/package-json-lint-config-terra/package-json-lint.config.js

```diff
@@ -2,5 +2,6 @@ module.exports = {
   rules: {
     'require-no-terra-base-peer-dependency-versions': 'warn',
     'require-theme-context-versions': 'warn',
+    'require-no-hard-coded-dependency-versions': 'warn',
```

Comment:
I actually think we should start this as an error out of the gate.

Reply:
Updated here https://github.com/cerner/terra-toolkit/pull/704/commits/cc6d1e0cc26383d90b787febbf2eeec56d41464a.

But now the build will fail due to hardcoded dependencies in these packages:
[eslint-config-terra](https://github.com/cerner/terra-toolkit/blob/main/packages/eslint-config-terra/package.json#L38)
[jest-config-terra](https://github.com/cerner/terra-toolkit/blob/main/packages/jest-config-terra/package.json#L43)
[terra-function-testing](https://github.com/cerner/terra-toolkit/blob/main/packages/terra-functional-testing/package.json#L56)

Reply:
Ahh I see. I think in the long run, we'll want consumers to be able to allow for these types of situations. Do you think it would be possible to allow for something like this in the config?

  "package-json-lint": {
    "extends": "package-json-lint.config",
    "projectType": "devModule",
    "rules": {
      "require-no-hard-coded-dependency-versions": ["error", { "allowList": "eslint-plugin-react" }]
    }
  },

Reply:
What I am thinking is, we can have something like this:
```js
"require-no-hard-coded-dependency-versions": {severityType: "error",  allowList: ["eslint-plugin-react"] }
```
What do you think? Can we allow an object to be sent for each of the rules, with severityType. This is to maintain consistency in how the rules are being passed.

Reply:
Let's go with your suggestion. What I had suggested was something that eslint does, but I think we can go our own way with this.

Reply:
Updated here https://github.com/cerner/terra-toolkit/pull/704/commits/dfc30a5064aee33acfa4abd9f9fe89e768933f93